### PR TITLE
V4 speedup sprite creation

### PR DESCRIPTION
**Pull request by PatriceJiang**
ref https://github.com/cocos2d/cocos2d-x/issues/19907

### CODE REVIEW

**Comment by crazyhappygame** [comment]
Could you explain what was performance problem and how you improve it? (benchmarks)

**Comment by PatriceJiang** [comment]
Creating 40000 sprites takes 130ms in v4 and 28ms in v3.  (Tested on Linux, i5 3GHz)
This PR reduces time to 62ms when creating sprites with default shaders.

**Comment by crazyhappygame** [comment]
Could you explain where is speedup for this implementation?

2x improvment is not bad but anyway it is 2x slower than v3....

File: cocos/2d/CCSprite.cpp [hunk 1]
```diff
@@ -372,11 +381,48 @@ void Sprite::setVertexLayout()
...
+
+
 void Sprite::updateShaders(const char* vert, const char* frag)
 {
-    auto programState = new (std::nothrow) backend::ProgramState(vert, frag);
-    setProgramState(programState);
-    CC_SAFE_RELEASE_NULL(programState);
+    if(_defaulCommandInfo.ps != nullptr && vert == positionTextureColor_vert && frag == positionTextureColor_frag)
+    {
+        auto programState = _defaulCommandInfo.ps->clone();
```
Comment (PatriceJiang) [inline cocos/2d/CCSprite.cpp:390]: `ProgramState::clone` is faster than `ProgramState::ProgramState(const char*, const char *)`

**Comment by PatriceJiang** [comment]
There are more abstractions on ProgramState in v4, it is hard to achieve the same speed as v3, and we may improve it in the future.

File: cocos/2d/CCSprite.cpp (hunk 1 above)
Comment (crazyhappygame) [reply cocos/2d/CCSprite.cpp:390]: This is strange....
There is already in ProgramState::ProgramState(const char*, const char *) one cache
` _program = backend::ProgramCache::getInstance()->newProgram`
and with this PR we would like to have another cache on top of backend::ProgramCache?

Comment (PatriceJiang) [reply cocos/2d/CCSprite.cpp:390]: We have discussed the performance issue of ProgramCache yesterday, and we will fix that later.

Comment (PatriceJiang) [reply cocos/2d/CCSprite.cpp:390]: Since we will optimize ProgramCache, this PR is not needed.

**Pull request closed without merging**